### PR TITLE
vim-patch:8.1.{223,836,1732}

### PR DESCRIPTION
**Pull request by janlazo**
8.1.2037 has the leak so deal with it later.

### CODE REVIEW

File: src/nvim/ex_docmd.c [hunk 1]
```diff
@@ -5984,13 +5986,21 @@ char_u *get_user_cmd_addr_type(expand_T *xp, int idx)
 /*
  * Function given to ExpandGeneric() to obtain the list of user command names.
  */
-char_u *get_user_commands(expand_T *xp, int idx)
+char_u *get_user_commands(expand_T *xp FUNC_ATTR_UNUSED, int idx)
+  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
```
Comment (janlazo) [inline src/nvim/ex_docmd.c]: This function doesn't seem to modify anything if `get_cmdline_type` is pure.

**Comment by justinmk** [comment]
restarted bsd builds (hardware problem).

asan found a leak: https://gist.github.com/justinmk/ae3ab28f8699a201807be135a16a12be

```
    #3 0xcd7db1 in throw_exception /home/travis/build/neovim/neovim/build/../src/nvim/ex_eval.c:467:10
    #4 0xcd7727 in do_errthrow /home/travis/build/neovim/neovim/build/../src/nvim/ex_eval.c:326:7
    #5 0xc31b8a in do_one_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:2292:3
    #6 0xc198be in do_cmdline /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:621:20
    #7 0x126797d in nv_colon /home/travis/build/neovim/neovim/build/../src/nvim/normal.c:4610:18
    #8 0x12515b9 in normal_execute /home/travis/build/neovim/neovim/build/../src/nvim/normal.c:1131:3
    #9 0x18fdb45 in state_enter /home/travis/build/neovim/neovim/build/../src/nvim/state.c:69:26
    #10 0x120a1f9 in normal_enter /home/travis/build/neovim/neovim/build/../src/nvim/normal.c:463:3
...
```

File: src/nvim/ex_docmd.c (hunk 1 above)
Comment (justinmk) [reply src/nvim/ex_docmd.c]: modify what? This function is marked `FUNC_ATTR_PURE`, so it shouldn't modify anything.

**Review by justinmk — APPROVED** [review]

File: src/nvim/ex_docmd.c (hunk 1 above)
Comment (janlazo) [reply src/nvim/ex_docmd.c]: I marked it pure but the helper functions that it uses aren't explicitly declared as const/pure functions.

**Pull request merged**